lib/firmata.js

```diff
@@ -209,20 +211,28 @@ SYSEX_RESPONSE[CAPABILITY_RESPONSE] = function(board) {
 
   // Only create pins if none have been previously created on the instance.
   if (!board.pins.length) {
+    var pin = {};
     for (var i = 2, n = 0; i < board.currentBuffer.length - 1; i++) {
       if (board.currentBuffer[i] === 127) {
-        board.pins.push({
-          supportedModes: supportedModes(capability),
-          mode: undefined,
-          value: 0,
-          report: 1,
-        });
+        pin.supportedModes = supportedModes(capability);
```

Comment:
Ok, I see now. I still think this should be refactored, to avoid this strange `pin` binding reassignment, but I think we can come back to it later (ie. not this PR).